src/zenml/orchestrators/step_runner.py

```diff
@@ -152,6 +152,22 @@ def run(
                 func=step_instance.entrypoint
             )
 
+            for k, v in list(output_annotations.items()):
+                if v.artifact_config:
+                    _evaluated_name = v.artifact_config._evaluated_name(
```

Comment:
Yeah, but I replaced it all the way down just to preserve the order of the keys and make this process a bit simpler. I can also p[reserve the keys order by other means, if you feel that it is important not to replace static entries.

Reply:
Mhm not sure I get what you mean by preserving the order here. Doesn't the current code remove all items for which `v.artifact_config.name is not None` and the others stay in the dict, so the order gets modified?

Reply:
Since 3.7 the dict keys order is guaranteed and we were relying on that. 
Let me try to explain by example:
- say your output is `["a1","a_{time}","a2"]` - dynamic is sitting between static ones
- you return `return 1,42,2` and expect `a1=1, a_10:40:20=42, a2=2`
- in the loop you are referring to now we do:
  - evaluate `a1` to `a1`, pop and insert - the dict keys are `a_{time},a2,a1` now
  - `a_{time}` to `a_10:40:20` - the dict keys are `a2,a1,a_10:40:20`
  - `a2` is `a2` and the dict keys become `a1,a_10:40:20,a2`
- All your artifacts get desired values after the step finishes

Now the same, but without poping and reassigning the static ones:
- Only `a_{time}` gets evaluated to something new, so the dict keys would be `a1,a2,a_10:40:20`
- Values assigned to `a2` and `a_10:40:20` would be mixed with one another

For that reason I always pop and reassign. Does it make sense?

Reply:
Yes absolutely, but what if `v.artifact_config is None`, or `evaluated_name is None`? Shouldn't this be happening in all cases to make sure the order is preserved?

Reply:
Reworked this piece and extended the tests set